bot/bot.py

```diff
@@ -92,8 +92,8 @@ def clear(self) -> None:
         self._recreate()
         super().clear()
 
-    def remove_extensions(self) -> None:
-        """Remove all extensions and Cog to close bot. Copy from discord.py's own `close` for right closing order."""
+    def _remove_extensions(self) -> None:
+        """Remove all extensions and Cog to close bot. Copyright (c) 2015-2020 Rapptz (discord.py, MIT License)."""
```

Comment:
This does not comply with the license. You need to include the copyright AND the permission notice. So basically, copy pasting the entire license. Again, look at the example I linked.

Reply:
I found that this don't actually need remove cogs manually: they get removed automatically by unloading extensions.

Reply:
Yes, but cogs aren't restricted to being defined in extensions. Even though (I think) they always are in our bot, the previous code was the proper approach.

Reply:
This is removing other cogs that is not in extensions later.

Reply:
Where does it remove those other cogs? This code was responsible for it, but you removed it

```py
for cog in tuple(self.cogs):
    try:
        self.remove_cog(cog)
    except Exception:
        pass```

Reply:
Now `await super().close()` remove other cogs too.